### DISCUSSION TITLE
Progressive Web App, Android & iOS app

### DISCUSSION THREAD

**Discussion by bil-ash** [misskey-dev/misskey#8402] [Ideas]
Although misskey serves both a manifest and service worker, it is not a progressive web app(PWA) (at least as of version 12.108) . This is because the service worker is registered only after a user sign-in is complete and does not cache any page although [almost entire site](https://misskey-hub.net/en/docs/admin/cdn.html#caching) is static . The service worker is used only for push notifications.
 Making it a PWA will enable easy android, iOS and windows app generation using https://www.pwabuilder.com which is not possible now as the pwabuilder crawler is not able to detect the service worker(because it acesses the site as a non signed-in user). 
![sw](https://user-images.githubusercontent.com/33216854/158022652-8ce328f8-a65a-4597-91f1-9b7791667036.png)
The admin panel can be changed to convert the given option(in pic) into two different ones-
a)Make a PWA
b)Enable push notifictions
Enabling (a) will be a prerequisite for enabling (b). At the same time. this will allow a user to generate a PWA without notifications by enabling (a) but disabling (b).

**Comment by ThatOneCalculator** [comment]
Also see #8437
